### PR TITLE
Consolidate Move resolver implementations

### PR DESCRIPTION
**Pull request by msmouse**
## Motivation

aptos-vm/src/data_cache.rs is now the single place where we adapt the StateView into the Move resolvers.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/aptos-labs/aptos-core/blob/main/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan
Existing coverage

### CODE REVIEW

File: api/types/src/convert.rs [hunk 1]
```diff
@@ -21,30 +21,35 @@ use move_binary_format::file_format::FunctionHandleIndex;
...
+impl<S: StateView> MoveConverter<S> {
+    pub fn new(state_view: S) -> Self {
         Self {
-            inner: MoveValueAnnotator::new(inner),
+            resolver: state_view.into_move_resolver(),
         }
     }
 
+    pub fn annotator(&self) -> MoveValueAnnotator<RemoteStorageOwned<S>> {
+        MoveValueAnnotator::new(&self.resolver)
```
Comment (zekun000) [inline api/types/src/convert.rs:50]: this is very expensive, there's a module cache lives inside the annotator

Comment (msmouse) [reply api/types/src/convert.rs:50]: The `MoveConverter` is reconstructed for every call anyway..  but what's the module you used to do internal refs again?

Comment (zekun000) [reply api/types/src/convert.rs:50]: ouroboros, but I think I tried it doesn't work with async context. there's Rc inside

Comment (msmouse) [reply api/types/src/convert.rs:50]: Okay it's ouroboros.. and it's tricky to get compiled..  I'll land de hopefully follow up later. Thank you.

File: storage/storage-interface/src/lib.rs [hunk 2]
```diff
@@ -322,12 +322,25 @@ pub trait DbReader: Send + Sync {
         unimplemented!()
     }
 
-    /// Returns the latest ledger info.
-    fn get_latest_ledger_info(&self) -> Result<LedgerInfoWithSignatures> {
+    /// Returns the latest ledger info, if any.
+    fn get_latest_ledger_info_option(&self) -> Result<Option<LedgerInfoWithSignatures>> {
```
Comment (sitalkedia) [inline storage/storage-interface/src/lib.rs:326]: Why did this change?

Comment (msmouse) [reply storage/storage-interface/src/lib.rs:326]: To power `DbStateView::is_genesis()`

**Comment by msmouse** [comment]
@zekun000  @sitalkedia  so I decided to make one more call on the callsite, so that I don't need to change `MoveConverter`

**Review by sitalkedia — APPROVED** [review]
LGTM

**Pull request merged**